pkg/crc/cluster/cluster.go

```diff
@@ -173,14 +173,24 @@ func RemovePullSecretFromCluster(ocConfig oc.Config, sshRunner *ssh.Runner) erro
 		return fmt.Errorf("Failed to remove Pull secret %w: %s", err, stderr)
 	}
 
-	cmdArgs = []string{"delete", "machineconfigs", "--all"}
-	_, stderr, err = ocConfig.RunOcCommand(cmdArgs...)
+	// This block should be removed as soon as we start shipping with 4.8 bundle.
+	// This check is only make sure if there is any machineconfig resource or not because
+	// in 4.7 we disabled mco and also deleted all the machineconfig/machineconfig pools.
+	// For 4.8 we don't disable mco and it does contain the machineconfigs.
+	cmd := "oc --kubeconfig=/opt/kubeconfig get mc --sort-by=.metadata.creationTimestamp --no-headers -oname | grep rendered-master"
+	_, _, err = sshRunner.Run(cmd)
+	if err != nil {
+		return nil
+	}
+
+	cmd = "oc --kubeconfig=/opt/kubeconfig get mc --sort-by=.metadata.creationTimestamp --no-headers -oname | grep rendered-master | head -n -1 | xargs -t oc --kubeconfig=/opt/kubeconfig delete"
+	_, stderr, err = sshRunner.Run(cmd)
```

Comment:
Did you try to go through an `OcRunner`? As I said before, 4.8 fixes to bundle generation can probably go in a separate PR so that we can move forward more quickly with the main 4.8 changes.